### ISSUE TITLE
Rerender

### ISSUE THREAD

**Issue by ayepRahman** [mpontus/react-modal-hook#8]
Hi there, I face a rerender of modal with the second arg state changes. I am using redux for state management and passing the second argument loading, but whenever state changes, The modal rerender. Is this an expected behavior?

**Comment by mpontus** [comment]
I think I know what you mean. To make a long story short, we render modal components, that you pass to `useModal` functions, directly to the dom. We use them as the components, so whenever their identity changes, as it happens on inputs update, [React reconciliation mechanism](https://reactjs.org/docs/reconciliation.html) would unmount the previous version of the component and mount it anew. This does not play well with things like Transition group, as well as causing a remount on all of the children components.

I tried to address it by introducing an immutable component as a proxy to modal rendering. To make it possible to inline the call to the modal component, they needed to be restricted to functional immutable components. Since it was previously possible, though not advertised, to pass class components to `useModal`, this could cause a breaking change for some users, which is why I published my changes with a new major version: [`react-modal-hook@2.0.0`](https://www.npmjs.com/package/react-modal-hook/v/2.0.0).

Also, I won't be able to test this change on a large project until after the next week, and until then I won't be updating the `latest` tag. So if you want to start using the updated version now you would need to install `react-modal-hook@next`, which I invite you to do and let me know if it fixes your issue.
